### DISCUSSION TITLE
Accessing parent context two parents up

### DISCUSSION THREAD

**Discussion by Matt-Cam** [Netflix/dgs-framework#330] [Q&A] [answered]
Hi, I have a question involving accessing a parent's parent context (two layers up) from within a datafetcher/dataloader. 

Suppose we have the following contrived example:
```
type A {
  value: Int
  some_child: B
}
type B {
  value: Int
  some_child: C
}
type C {
  sumOfParentValueAndItsParentValue: Int // suppose this int is formed via parentA.value + parentB.value
}
```
I understand that type C can access type B via a call to dfe.getSource(). Now, what is the standard practice of accessing parent A from type C (which is nested two layers below A)? Additionally is this sort of situation of needing to access two parent levels up an indication of faulty/poor schema design?

**Comment by paulbakker** [comment, marked as answer]
Hi @SirMattCam, we have a guide about that! :-)

https://netflix.github.io/dgs/advanced/context-passing/#no-showid-use-local-context

**Reply by Matt-Cam** [reply]
Thanks @paulbakker! Now in that example, the shows() DataFetcher is returning a list of shows as well as setting a map of InternalShows on the local context. My follow up question there would be, how would this change if the shows DataFetcher also made use of a DataLoader? Would the signature be one of the following?
```public CompletableFuture<DataFetcherResult<List<Show>>> shows(...) {...```
```public DataFetcherResult<CompletableFuture<List<Show>>> shows(...) {...```
